src/collector.rs

```diff
@@ -51,6 +51,19 @@ impl Collector {
         self.raw.get(command).unwrap().clone()
     }
 
+    // returns a string representation of the first capture group of the regex
+    pub fn get_regex_from_command(&mut self, command: &str, regex: &str) -> Option<String> {
+        let re = Regex::new(regex).unwrap();
```

Comment:
This is allowed to panic, our regexes should always at least compile (assuming that's what the possible error is here)